reference_test.go

```diff
@@ -199,8 +199,6 @@ func checkRefType(t *testing.T, ref *Reference, kind ReferenceType) {
 	// The failure happens at wherever we were called, not here
 	_, file, line, ok := runtime.Caller(1)
 	if !ok {
-		t.Fatal()
+		t.Fatalf("Wrong ref type at %v:%v; have %v, expected %v", file, line, ref.Type(), kind)
 	}
-
-	t.Fatalf("Wrong ref type at %v:%v; have %v, expected %v", file, line, ref.Type(), kind)
```

Comment:
This is correct. The `!ok` block is for the case where we are unable to access the stack. This change would make it so `checkRefType` would never assert the type of ref.